src/games/strategy/triplea/ui/screen/IDrawable.java

```diff
@@ -180,6 +183,57 @@ public void draw(final Rectangle bounds, final GameData data, final Graphics2D g
     }
   }
 
+  /**
+   * Find the best rectangle inside the territory to place the name in. Finds the rectangle
+   * that can fit the name, that is the closest to the vertical center, and has a large width at
+   * that location. If there isn't any rectangles that can fit the name then default back to the
+   * bounding rectangle.
+   */
+  private Rectangle getBestTerritoryNameRect(final MapData mapData, final Territory territory,
+      final FontMetrics fontMetrics) {
+
+    // Find bounding rectangle and use to create parameters for a grid across the territory
+    final Rectangle territoryBounds = mapData.getBoundingRect(territory);
+    Rectangle result = territoryBounds;
+    final int maxX = territoryBounds.x + territoryBounds.width;
+    final int maxY = territoryBounds.y + territoryBounds.height;
+    final int centerY = territoryBounds.y + territoryBounds.height / 2;
+    final int minWidth = fontMetrics.stringWidth(territory.getName());
+    final int increment = fontMetrics.getAscent();
+    int maxScore = 0;
+
+    // Loop through the grid moving the starting point and determining max width at that point
+    for (int leftX = territoryBounds.x; leftX < maxX - increment; leftX += increment) {
+      for (int topY = territoryBounds.y; topY < maxY - increment; topY += increment) {
```

Comment:
You mentioned the score is maximized when Y is half way through the loop. That should be made clear in the code. Right now a person has to unwind all variables back to their `territoryBounds` definitions, which gets pretty gnarly very readily.